third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -361,6 +361,21 @@ void createBarrier(ConversionPatternRewriter &rewriter, Location loc,
 }
 } // namespace
 
+static LLVM::AtomicOrdering llvmMemoryOrdering(MemSemantic memOrdering) {
+  const static std::map<MemSemantic, LLVM::AtomicOrdering> memOrderingMap = {
```

Comment:
Normally we don't want to have non-POD `static` to avoid complexity. STL containers have complicated initialization logic that would need to happen, and we need to guarantee the initialization once semantics regarding multithreading and such. You can see other issues in discussions like https://softwareengineering.stackexchange.com/questions/350501/is-there-any-benefit-to-to-define-constant-local-variables-as-static-c.  It's a bit overkill for this usecase, where we just have 4 elements (and unlikely to grow in the future). What about just using a `switch` statement here?

Reply:
Thanks for the comments. I used the static so avoid the initialization each time the function is called. As you said, there are only 4 cases here, I change to `switch` as you suggested.